Library/Homebrew/extend/string.rb

```diff
@@ -10,4 +10,16 @@ class String
   #   "hello".exclude? ?h   # => false
   sig { params(string: String).returns(T::Boolean) }
   def exclude?(string) = !include?(string)
+
+  # Converts a string starting with `:` to a symbol, otherwise returns the
+  # string itself.
+  #
+  #   ":example".to_string_or_symbol # => :example
+  #   "example".to_string_or_symbol  # => "example"
+  sig { returns(T.any(String, Symbol)) }
+  def to_string_or_symbol
```

Comment:
Would rather we avoid a global monkey patch here. Instead let's add to `Utils` somewhere.